### PR TITLE
Several regressions over time

### PR DESCRIPTION
**Pull request by marudor**
I've started to retest all libdefs and started to fix or limit libdefs that got broken over time.

### CODE REVIEW

**Comment by villesau** [comment]
@marudor looks like `test_react-apollo_v2.x.x.js` is conflicting.

**Comment by pascalduez** [comment]
There's a bunch of "weird" formatting changes, lots of whitespace in indentation added, is it intended?

File: definitions/npm/awilix_v2.x.x/flow_v0.56.x-v0.70.x/awilix_v2.x.x.js [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-    AwilixResolutionError: awilix$ResolutionError,
-    listModules: awilix$listModules
+    createContainer(?awilix$ContainerOptions): awilix$Container < {} >,
+      ResolutionMode: Class < awilix$ResolutionMode >,
+        Lifetime: Class < awilix$Lifetime >,
+          asValue: awilix$asValue,
+            asClass: awilix$asClass,
+              asFunction: awilix$asFunction,
+                AwilixResolutionError: awilix$ResolutionError,
+                  listModules: awilix$listModules
```
Comment (villesau) [inline definitions/npm/awilix_v2.x.x/flow_v0.56.x-v0.70.x/awilix_v2.x.x.js]: hmm yeah, these are a bit weird?

File: definitions/npm/algoliasearch_v3.x.x/flow_v0.71.x-/algoliasearch_v3.x.x.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  hits: Array < any >,
+    page: number,
+      nbHits: number,
+        nbPages: number,
+          hitsPerPage: number,
+            processingTimeMS: number,
+              query: string,
+                parsed_query: string,
+                  params: string
+                    |};
```
Comment (villesau) [inline definitions/npm/algoliasearch_v3.x.x/flow_v0.71.x-/algoliasearch_v3.x.x.js]: There is still quite a lot of weird formatting. Maybe worth running e.g trough prettier?

**Comment by marudor** [comment]
I've run prettier for all my changes. Formatting should be okay now.

**Comment by villesau** [comment]
@marudor  Looks like there are still some slight errors, maybe due to latest flow release: https://travis-ci.org/flow-typed/flow-typed/jobs/448255030

**Comment by AndrewSouthpaw** [comment]
Everything worked except for `icepick`:

```

ERROR: icepick_v1.x.x/flow_v0.47.x-v0.73.x
 * icepick_v1.x.x/flow_v0.47.x-v0.73.x (flow-v0.73.0): Unexpected Flow errors(2):
   Error ------------------------------------------------------------------------------------ test_icepick_v1.x.x.js:277:14

   Cannot call `i.chain(...).assoc` with `"key"` bound to `path` because string [1] is incompatible with number [2].

      test_icepick_v1.x.x.js:277:14
      277|       .assoc("key", 5)
                        ^^^^^ [1]

   References:
      icepick_v1.x.x.js:138:17
      138|     assoc(path: number, value: T): IcepickArrayWrapper<T>;
                           ^^^^^^ [2]


   Error ------------------------------------------------------------------------------------ test_icepick_v1.x.x.js:293:12

   Cannot call `i.chain(...).set` with `"key"` bound to `key` because string [1] is incompatible with number [2].

      test_icepick_v1.x.x.js:293:12
      293|       .set("key", 5)
                      ^^^^^ [1]

   References:
      icepick_v1.x.x.js:139:14
      139|     set(key: number, value: T): IcepickArrayWrapper<T>;
                        ^^^^^^ [2]


   Error ------------------------------------------------------------------------------------ test_icepick_v1.x.x.js:309:15

   Cannot call `i.chain(...).dissoc` with `"key"` bound to `key` because string [1] is incompatible with number [2].

      test_icepick_v1.x.x.js:309:15
      309|       .dissoc("key")
                         ^^^^^ [1]

   References:
      icepick_v1.x.x.js:140:17
      140|     dissoc(key: number): IcepickArrayWrapper<T>;
                           ^^^^^^ [2]


   Error ------------------------------------------------------------------------------------ test_icepick_v1.x.x.js:325:14

   Cannot call `i.chain(...).unset` with `"key"` bound to `key` because string [1] is incompatible with number [2].

      test_icepick_v1.x.x.js:325:14
      325|       .unset("key")
                        ^^^^^ [1]

   References:
      icepick_v1.x.x.js:141:16
      141|     unset(key: number): IcepickArrayWrapper<T | null>;
                          ^^^^^^ [2]


   Error ------------------------------------------------------------------------------------ test_icepick_v1.x.x.js:345:13

   Cannot call `i.chain(...).thru` with function bound to the first parameter because number [1] is incompatible with array
   type [2] in the return value.

      test_icepick_v1.x.x.js:345:13
                       v--------------
      345|       .thru(function(val) {
      346|         return 5;
      347|       })
                 ^

   References:
      test_icepick_v1.x.x.js:346:16
      346|         return 5;
                          ^ [1]
      icepick_v1.x.x.js:156:24
      156|     thru((val: T[]) => T[]): IcepickArrayWrapper<T>;
                                  ^^^ [2]


   Warning ----------------------------------------------------------------------------------- test_icepick_v1.x.x.js:275:5

   Unused suppression comment.

      275|     // $ExpectError
               ^^^^^^^^^^^^^^^


   Warning ----------------------------------------------------------------------------------- test_icepick_v1.x.x.js:291:5

   Unused suppression comment.

      291|     // $ExpectError
               ^^^^^^^^^^^^^^^


   Warning ----------------------------------------------------------------------------------- test_icepick_v1.x.x.js:307:5

   Unused suppression comment.

      307|     // $ExpectError
               ^^^^^^^^^^^^^^^


   Warning ----------------------------------------------------------------------------------- test_icepick_v1.x.x.js:323:5

   Unused suppression comment.

      323|     // $ExpectError
               ^^^^^^^^^^^^^^^


   Warning ----------------------------------------------------------------------------------- test_icepick_v1.x.x.js:343:5

   Unused suppression comment.

      343|     // $ExpectError
               ^^^^^^^^^^^^^^^



   Found 5 errors and 5 warnings

   null

 * icepick_v1.x.x/flow_v0.47.x-v0.73.x (flow-v0.72.0): Unexpected Flow errors(2):
   Error ------------------------------------------------------------------------------------ test_icepick_v1.x.x.js:277:14

   Cannot call `i.chain(...).assoc` with `"key"` bound to `path` because string [1] is incompatible with number [2].

      test_icepick_v1.x.x.js:277:14
      277|       .assoc("key", 5)
                        ^^^^^ [1]

   References:
      icepick_v1.x.x.js:138:17
      138|     assoc(path: number, value: T): IcepickArrayWrapper<T>;
                           ^^^^^^ [2]


   Error ------------------------------------------------------------------------------------ test_icepick_v1.x.x.js:293:12

   Cannot call `i.chain(...).set` with `"key"` bound to `key` because string [1] is incompatible with number [2].

      test_icepick_v1.x.x.js:293:12
      293|       .set("key", 5)
                      ^^^^^ [1]

   References:
      icepick_v1.x.x.js:139:14
      139|     set(key: number, value: T): IcepickArrayWrapper<T>;
                        ^^^^^^ [2]


   Error ------------------------------------------------------------------------------------ test_icepick_v1.x.x.js:309:15

   Cannot call `i.chain(...).dissoc` with `"key"` bound to `key` because string [1] is incompatible with number [2].

      test_icepick_v1.x.x.js:309:15
      309|       .dissoc("key")
                         ^^^^^ [1]

   References:
      icepick_v1.x.x.js:140:17
      140|     dissoc(key: number): IcepickArrayWrapper<T>;
                           ^^^^^^ [2]


   Error ------------------------------------------------------------------------------------ test_icepick_v1.x.x.js:325:14

   Cannot call `i.chain(...).unset` with `"key"` bound to `key` because string [1] is incompatible with number [2].

      test_icepick_v1.x.x.js:325:14
      325|       .unset("key")
                        ^^^^^ [1]

   References:
      icepick_v1.x.x.js:141:16
      141|     unset(key: number): IcepickArrayWrapper<T | null>;
                          ^^^^^^ [2]


   Error ------------------------------------------------------------------------------------ test_icepick_v1.x.x.js:345:13

   Cannot call `i.chain(...).thru` with function bound to the first parameter because number [1] is incompatible with array
   type [2] in the return value.

      test_icepick_v1.x.x.js:345:13
                       v--------------
      345|       .thru(function(val) {
      346|         return 5;
      347|       })
                 ^

   References:
      test_icepick_v1.x.x.js:346:16
      346|         return 5;
                          ^ [1]
      icepick_v1.x.x.js:156:24
      156|     thru((val: T[]) => T[]): IcepickArrayWrapper<T>;
                                  ^^^ [2]


   Warning ----------------------------------------------------------------------------------- test_icepick_v1.x.x.js:275:5

   Unused suppression comment.

      275|     // $ExpectError
               ^^^^^^^^^^^^^^^


   Warning ----------------------------------------------------------------------------------- test_icepick_v1.x.x.js:291:5

   Unused suppression comment.

      291|     // $ExpectError
               ^^^^^^^^^^^^^^^


   Warning ----------------------------------------------------------------------------------- test_icepick_v1.x.x.js:307:5

   Unused suppression comment.

      307|     // $ExpectError
               ^^^^^^^^^^^^^^^


   Warning ----------------------------------------------------------------------------------- test_icepick_v1.x.x.js:323:5

   Unused suppression comment.

      323|     // $ExpectError
               ^^^^^^^^^^^^^^^


   Warning ----------------------------------------------------------------------------------- test_icepick_v1.x.x.js:343:5

   Unused suppression comment.

      343|     // $ExpectError
               ^^^^^^^^^^^^^^^



   Found 5 errors and 5 warnings

   null
```

**Comment by villesau** [comment]
Awesome work @marudor ! :)

**Pull request merged**